### PR TITLE
Add a generic DFU firmware object for plugins to use

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: src/fu-dfu-firmware.c [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	FuDfuFirmwarePrivate *priv = GET_PRIVATE (self);
+	g_return_val_if_fail (FU_IS_DFU_FIRMWARE (self), 0x0);
+	return priv->pid;
+}
+
+/**
+ * fu_dfu_firmware_get_release:
+ * @self: a #FuDfuFirmware
+ *
+ * Gets the device ID.
```
Comment (superm1) [inline src/fu-dfu-firmware.c]: Comment wrong

File: src/fu-dfu-firmware.c [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+{
+	FuDfuFirmwarePrivate *priv = GET_PRIVATE (self);
+	g_return_if_fail (FU_IS_DFU_FIRMWARE (self));
+	priv->vid = vid;
+}
+
+/**
+ * fu_dfu_firmware_set_pid:
+ * @self: a #FuDfuFirmware
+ * @pid: product ID, or 0xffff for unset
```
Comment (superm1) [inline src/fu-dfu-firmware.c]: Unset?

File: src/fu-dfu-firmware.c [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+{
+	FuDfuFirmwarePrivate *priv = GET_PRIVATE (self);
+	g_return_if_fail (FU_IS_DFU_FIRMWARE (self));
+	priv->pid = pid;
+}
+
+/**
+ * fu_dfu_firmware_set_release:
+ * @self: a #FuDfuFirmware
+ * @release: device ID, or 0xffff for unset
```
Comment (superm1) [inline src/fu-dfu-firmware.c]: Unset?

File: src/fu-dfu-firmware.c (hunk 2 above)
Comment (hughsie) [reply src/fu-dfu-firmware.c]: would `by default` be clearer?

Comment (superm1) [reply src/fu-dfu-firmware.c]: It's a set function, there should be no default

Comment (hughsie) [reply src/fu-dfu-firmware.c]: Maybe I'm just being unclear. In DFU the 0xffff value means "matches any vendor" -- maybe I should just say that?

Comment (superm1) [reply src/fu-dfu-firmware.c]: Yes that's better

**Comment by superm1** [comment]
Lgtm

**Pull request merged**